atc/engine/build_step_delegate.go

```diff
@@ -172,6 +172,22 @@ func (delegate *buildStepDelegate) Finished(logger lager.Logger, succeeded bool)
 	logger.Info("finished")
 }
 
+func (delegate *buildStepDelegate) BeforeSelectWorker(logger lager.Logger) error {
+	if delegate.build.Name() == db.CheckBuildName {
```

Comment:
Could we instead check `delegate.build.ResourceID() != 0`?

Reply:
Sure.